src/unit-taxonomy-tags-drawer/TaxonomyTagsCollapsible.jsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import {
+  Badge,
+  Collapsible,
+  SelectableBox,
+  Button,
+  ModalPopup,
+  useToggle,
+} from '@edx/paragon';
+import { injectIntl } from '@edx/frontend-platform/i18n';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import './TaxonomyTagsCollapsible.scss';
+
+import TaxonomyTagsDropDownSelector from './TaxonomyTagsDropDownSelector';
+import { useTaxonomyTagsDataResponse, useIsTaxonomyTagsDataLoaded } from './api/hooks/selectors';
+
+import TagBubble from './TagBubble';
+
+const TaxonomyTagsCollapsible = ({ taxonomyAndTagsData }) => {
+  const {
+    id, name, unitTags,
+  } = taxonomyAndTagsData;
+
+  const [isOpen, open, close] = useToggle(false);
+  const [target, setTarget] = React.useState(null);
+
+  const useTaxonomyTagsData = (taxonomyId, fullPathProvided) => {
+    const taxonomyTagsData = useTaxonomyTagsDataResponse(taxonomyId, fullPathProvided);
+    const isTaxonomyTagsLoaded = useIsTaxonomyTagsDataLoaded(taxonomyId, fullPathProvided);
+    return { taxonomyTagsData, isTaxonomyTagsLoaded };
+  };
+
+  const { taxonomyTagsData, isTaxonomyTagsLoaded } = useTaxonomyTagsData(id);
+
+  return (
+    <div className="taxonomy-tags-collapsible-div">
+      <Collapsible title={name} styling="card-lg" className="taxonomy-tags-collapsible">
+        <div key={id}>
+          {
+            unitTags.map((unitTag) => {
+              let level = -1; // Starting from -1 since we increment at beginning of loop
+              return unitTag.lineage.map((lineageTag) => {
+                level += 1;
+                return (<TagBubble key={`tag-${lineageTag}`} value={lineageTag} implicit={lineageTag !== unitTag.value} level={level} />);
```

Comment:
This is not rendering multiple tags properly. (BTW we should change the lightcast skills taxonomy to `allow_multiple=True` and make it apply to all orgs in the sample data.) I suggest that you make this component so that it accepts the existing object tags data for this single taxonomy as a prop, and it also has an "additional selected tags" as internal state (which will get implemented by the "Add Tags" button), then you create a function to combine these together into a tree:

```js
// appliedObjectTags is a prop, which has the data from the content_object_tags REST API, but only for this current taxonomy.
const [changedTags, setChangedTags] = React.useState(...);
const tagsTree = React.useMemo(() => {
    // compute the tree from appliedObjectTags, using the lineage
    // add in any newly added tags (that haven't yet been saved to the server) - can come in next PR
    // remove any deleted tags - can come in next PR
    // delete any "orphaned" parents with no children - can come in next PR
    return tagsTree;
}, [appliedObjectTags, changedTags]);
// render the view using tagsTree
```

**How this looks now with multiple tags:**

![Screenshot 2023-11-08 at 1 43 03 PM](https://github.com/openedx/frontend-app-course-authoring/assets/945577/8961c102-45c3-4cd9-9d76-0dd3f9aba2b3)

**How it should look:**

![Screenshot 2023-11-08 at 1 44 34 PM](https://github.com/openedx/frontend-app-course-authoring/assets/945577/7df30715-1de0-46cd-8480-f1dc0052ece8)

Reply:
Oh thanks for catching this! It makes sense to create a separate component to handle this logic.

> (BTW we should change the lightcast skills taxonomy to allow_multiple=True and make it apply to all orgs in the sample data.)

Yup, I agree, I'll update the sample data script and create a PR once I've addressed all the issues.

Reply:
@bradenmacdonald I've implemented this, and managed to fit it in as part of this PR https://github.com/openedx/frontend-app-course-authoring/commit/f8cbc9122e51797c2a25a0b9d6ac0f576356f446 . Your suggestions worked great, and sets up up code for the the editing PR that follows.